File: src/main/java/ru/citeck/ecos/records2/request/query/SortBy.java
package ru.citeck.ecos.records2.request.query;

public class SortBy {

    private String attribute;
    private boolean ascending;

    public SortBy() {
    }

    public SortBy(String attribute, boolean ascending) {
        this.attribute = attribute;
        this.ascending = ascending;
    }

    public String getAttribute() {
        return attribute;
    }

    public void setAttribute(String attribute) {
        this.attribute = attribute;
    }

    public void setAscending(boolean ascending) {
        this.ascending = ascending;
    }

    public boolean isAscending() {
        return ascending;
    }

    @Override
    public String toString() {
        return "SortBy{" +
                "attribute='" + attribute + '\'' +
                ", ascending=" + ascending +
                '}';
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/record/dao/query/dto/query/SortBy.kt
package ru.citeck.ecos.records3.record.dao.query.dto.query

import ecos.com.fasterxml.jackson210.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonDeserialize as JackJsonDeserialize

@JsonDeserialize(builder = SortBy.Builder::class)
@JackJsonDeserialize(builder = SortBy.Builder::class)
data class SortBy(
    val attribute: String,
    val ascending: Boolean
) {

    companion object {

        @JvmStatic
        fun create(): Builder {
            return Builder()
        }

        @JvmStatic
        fun create(builder: Builder.() -> Unit): SortBy {
            val builderObj = Builder()
            builder.invoke(builderObj)
            return builderObj.build()
        }
    }

    fun copy(): Builder {
        return Builder(this)
    }

    fun copy(builder: Builder.() -> Unit): SortBy {
        val builderObj = Builder(this)
        builder.invoke(builderObj)
        return builderObj.build()
    }

    class Builder() {

        var attribute: String = ""
        var ascending: Boolean = false

        constructor(base: SortBy) : this() {
            attribute = base.attribute
            ascending = base.ascending
        }

        fun build(): SortBy {
            return SortBy(attribute, ascending)
        }
    }
}
